src/inputs/EditableLabel/EditableLabel.js

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { TextField, makeStyles, Typography, CircularProgress } from '@material-ui/core';
+import EditIcon from '@material-ui/icons/Edit';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    height: '100%',
+  },
+  labelDisplayContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'flex-start',
+    alignItems: 'stretch',
+  },
+  label: {
+    maxWidth: '500px',
+    color: theme.palette.primary.main,
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+  },
+  textField: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'flex-start',
+    alignItems: 'stretch',
+    '& .MuiOutlinedInput-input': {
+      borderWidth: '2px',
+      outline: 'none',
+      padding: '8px',
+    },
+    '& .MuiOutlinedInput-root': {
+      minWidth: '300px',
+    },
+  },
+  editIcon: {
+    marginRight: '10px',
+    fontWeight: 'bold',
+    height: '20px',
+    color: theme.palette.primary.main,
+  },
+  circularLoading: {
+    marginLeft: '15px',
+    color: theme.palette.primary.contrastText,
+    backgroundColor: theme.palette.background.card,
+  },
+}));
+
+export const EditableLabel = (props) => {
+  const { value, onProcess, checkValue, labels } = props;
+
+  const classes = useStyles();
+
+  const [editMode, setEditMode] = useState(false);
```

Comment:
I know we're using `editMode` in the webapp, but we could improve the name of this variable by following a common convention on boolean variables names : bool variables should start with `is` or `has` (or `can`, `should`, ... ).
Could we rename this state to `isEditing` and `setEditing` ?